### PR TITLE
Add project mode to REST API and project files

### PR DESCRIPTION
**Pull request by J535D165**

### CODE REVIEW

File: asreview/webapp/src/PreReviewComponents/ProjectInit.js [hunk 1]
```diff
@@ -0,0 +1,237 @@
+import React from "react";
```
Comment (J535D165) [inline asreview/webapp/src/PreReviewComponents/ProjectInit.js:1]: @terrymyc was this file moved on master?

Comment (terrymyc) [reply asreview/webapp/src/PreReviewComponents/ProjectInit.js:1]: This file was renamed as `ProjectInfo.js`.

File: asreview/webapp/src/PreReviewComponents/ProjectInfo.js [hunk 2]
```diff
@@ -96,26 +110,17 @@ const ProjectInfo = (props) => {
...
     var bodyFormData = new FormData();
+    bodyFormData.set("mode", info.mode);
     bodyFormData.set("name", info.name);
     bodyFormData.set("authors", info.authors);
     bodyFormData.set("description", info.description);
 
-    (props.edit
-      ? ProjectAPI.info(props.project_id, true, bodyFormData)
-      : ProjectAPI.init(bodyFormData)
-    )
```
Comment (terrymyc) [inline asreview/webapp/src/PreReviewComponents/ProjectInfo.js:106]: All lines involving the removal of `props.edit` will affect the functionality of editing the project info. Currently, this initiating a new project function and editing the project info function share the same dialog. For now, I would propose to keep `props.edit` as an indicator of whether this dialog is used as a function to initiate or edit a project. And when the dialog plays the function of editing the project info, no selection menu for the mode is displayed.
In the future, I expect these two functions to be separated. The function to initiate a project should be integrated into the setup stage.

**Review by terrymyc — CHANGES_REQUESTED** [review]

File: asreview/webapp/api.py [hunk 3]
```diff
@@ -714,49 +703,112 @@ def api_set_algorithms(project_id):  # noqa: F401
...
+            logging.error(err)
+            return jsonify(message="Failed to train the model."), 500
 
     response = jsonify({'success': True})
     response.headers.add('Access-Control-Allow-Origin', '*')
     return response
 
 
-@bp.route('/project/<project_id>/model/init_ready', methods=["GET"])
+@bp.route('/project/<project_id>/ready', methods=["GET"])
```
Comment (terrymyc) [inline asreview/webapp/api.py:762]: The corresponding test for this API in `test_api.py` should also be changed.

**Review by terrymyc — CHANGES_REQUESTED** [review]

File: asreview/webapp/src/PreReviewComponents/ProjectInfo.js (hunk 2 above)
Comment (J535D165) [reply asreview/webapp/src/PreReviewComponents/ProjectInfo.js:106]: Thanks, great remark. I will try to resolve these conflicts and make the edit function work again.

Comment (J535D165) [reply asreview/webapp/src/PreReviewComponents/ProjectInfo.js:106]: I think I resolved this.

File: asreview/webapp/api.py (hunk 3 above)
Comment (J535D165) [reply asreview/webapp/api.py:762]: Thanks

**Comment by J535D165** [comment]
I'm merging this even though the tests fail. These tests are designed to test old project files for forward compatibility. They fail for various reasons (not sure what the cause is). However, I want to avoid more merge conflicts and merge this PR. 

Another remark: the simulation mode doesn't work properly yet (seems to be random). Working on this in a new PR.

**Pull request merged**